From 9675921fc8ad7864b22744bc67544824d1833a2b Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Mon, 20 Feb 2012 17:41:55 +0530
Subject: [PATCH 094/186] MTD: OMAP2+: elm: Initialize ELM module when needed

This patch configures ELM module for BCH8 ECC scheme on platforms where it is
being used.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 drivers/mtd/nand/omap2.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/mtd/nand/omap2.c b/drivers/mtd/nand/omap2.c
index 5fb710f..75f347e 100644
--- a/drivers/mtd/nand/omap2.c
+++ b/drivers/mtd/nand/omap2.c
@@ -24,6 +24,7 @@
 #include <plat/dma.h>
 #include <plat/gpmc.h>
 #include <plat/nand.h>
+#include <plat/elm.h>
 
 #define	DRIVER_NAME	"omap2-nand"
 #define	OMAP_NAND_TIMEOUT_MS	5000
@@ -974,6 +975,14 @@ static int __devinit omap_nand_probe(struct platform_device *pdev)
 	info->nand.options	= pdata->devsize;
 	info->nand.options	|= NAND_SKIP_BBTSCAN;
 
+	/*
+	 * If ELM feature is used in OMAP NAND driver, then configure it
+	 */
+	if (pdata->elm_used) {
+		if (pdata->ecc_opt == OMAP_ECC_BCH8_CODE_HW)
+			omap_configure_elm(&info->mtd, OMAP_BCH8_ECC);
+	}
+
 	/* NAND write protect off */
 	gpmc_cs_configure(info->gpmc_cs, GPMC_CONFIG_WP, 0);
 
-- 
1.7.9.1

